### ISSUE TITLE
[PHP] Accept / Return strings for Enum Constants

### ISSUE THREAD

**Issue by bshaffer** [protocolbuffers/protobuf#4474]
Protobuf should accept strings for Enum properties and return strings from Enum Constants. For example, [`PublicKeyCertificate::setFormat`](https://github.com/googleapis/proto-client-php/blob/c2d9d662c344db31ee793d1ad3c03827d225ee48/src/Google/Cloud/Iot/V1/PublicKeyCertificate.php#L60) should accept the strings `'unspecified'`, `'RSA_X509_PEM'`, `'ES256_PEM'`, etc, instead of requiring integers, e.g. `PublicKeyFormat::RSA_X509_PEM`:

```php
// current
$publicKey = (new PublicKeyCredential())
    ->setFormat(PublicKeyFormat::RSA_X509_PEM)
    ->setKey(file_get_contents($certificateFile));

// new
$publicKey = (new PublicKeyCredential())
    ->setFormat('RSA_X509_PEM')
    ->setKey(file_get_contents($certificateFile));
```

Similarly, for getters, we should provide an additional method (such as `getFormatString`) which returns the string representation of the constant:

```php
// current
$formats = [
    PublicKeyFormat::UNSPECIFIED_PUBLIC_KEY_FORMAT => 'unspecified',
    PublicKeyFormat::RSA_X509_PEM => 'RSA_X509_PEM',
    PublicKeyFormat::ES256_PEM => 'ES256_PEM',
    PublicKeyFormat::RSA_PEM => 'RSA_PEM',
    PublicKeyFormat::ES256_X509_PEM => 'ES256_X509_PEM',
];
foreach ($device->getCredentials() as $credential) {
    echo 'Format is ' . $formats[$credential->getPublicKey()->getFormat()];
}

// new
foreach ($device->getCredentials() as $credential) {
    echo 'Format is ' . $credential->getPublicKey()->getFormatString();
}
```

**Comment by bshaffer** [comment]
This is currently implemented for 
 - [NodeJS](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/iot/manager/manager.js#L345)
 - [Ruby](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/iot/iot.rb#L246)
 - [Python](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/iot/api-client/manager/manager.py#L101)
 - [Java](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/iot/api-client/manager/src/main/java/com/example/cloud/iot/examples/DeviceRegistryExample.java#L223)
 - [Go](https://github.com/GoogleCloudPlatform/golang-samples/blob/master/iot/manager/manager.go#L246)
 -  [C#](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/master/iot/api/Program.cs#L495)

**Comment by TeBoring** [comment]
Does the project you mentioned use protobuf for these languages?

**Comment by TeBoring** [comment]
And why don't you use string field in that case?

**Comment by bshaffer** [comment]
> And why don't you use string field in that case?

Because the field is actually a proto enum.

Another option is to go the python route and add a "name" and "value" method on the enum class that will do the conversion (see [here](https://developers.google.com/protocol-buffers/docs/reference/python-generated#enum))

**Comment by xfxyjwf** [comment]
@bshaffer I think you must be mistaken. The `PublicKeyCredential` type in Java you point to is simply not a protobuf type. It's likely also the case for your other pointers.

**Comment by TeBoring** [comment]
Fixed in #4534
